### PR TITLE
Clarify license agreement when setting OSX_SDK_DIR

### PR DESCRIPTION
**Pull request by matthiasdiener**
Reverts #1531 

Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

**Comment by isuruf** [comment]
I'd rather that we change the error message to say that by setting the env variable, you agree to the licensing and terms of the SDK by Apple.

**Comment by matthiasdiener** [comment]
Would you suggest removing this code in `setup_environment` then:


https://github.com/conda-forge/conda-smithy/blob/81e781f38b7559f8c95c412be051d5902df2939f/conda_smithy/feedstock_content/build-locally.py#L25-L28

**Comment by isuruf** [comment]
Yes

**Comment by matthiasdiener** [comment]
> Would you suggest removing this code in `setup_environment` then:
> 
> https://github.com/conda-forge/conda-smithy/blob/81e781f38b7559f8c95c412be051d5902df2939f/conda_smithy/feedstock_content/build-locally.py#L25-L28

Another option might be to add a license warning here instead of removing the code.

File: conda_smithy/feedstock_content/build-locally.py [hunk 1]
```diff
@@ -68,8 +64,9 @@ def verify_config(ns):
     elif ns.config.startswith("osx") and platform.system() == "Darwin":
```
Comment (isuruf) [inline conda_smithy/feedstock_content/build-locally.py:64]:
```suggestion
    elif ns.config.startswith("osx"):
```

**Comment by jakirkham** [comment]
cc @183amir (for vis)

**Comment by 183amir** [comment]
I don't use this script anymore. So I'm fine with any changes.

**Pull request merged**